src/client/ui/constructor/constructor.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect, useMemo } from 'https://esm.sh/react@18.2.0'
+import { API_URL } from '../../../const.ts'
+
+export const Constructor = () => {
+  const [iconOptions, setIconOptions] = React.useState<string[]>([])
+  const [badgeType, setBadgeType] = React.useState('plain')
+  const [plainFields, setPlainFields] = React.useState({
+    title: '',
+    color: '',
+    icon: '',
+    value: '',
+  })
+  const [skillFields, setSkillFields] = React.useState({
+    title: '',
+    icon: '',
+    value: '',
+  })
+  const [stackItems, setStackItems] = React.useState([
+    {
+      title: '',
+      icon: '',
+      value: '',
+    },
+  ])
+  const [showIconDropdown, setShowIconDropdown] = React.useState(false)
+  const [stackIconDropdown, setStackIconDropdown] = React.useState([false])
+  const [filteredIcons, setFilteredIcons] = React.useState(iconOptions)
+  const [resultUrl, setResultUrl] = React.useState('')
+  const [showCopied, setShowCopied] = React.useState(false)
+
+  useEffect(() => {
+    const fetchIcons = async () => {
+      const icons = await fetch(`${API_URL}/icons`).then(res => res.json())
```

Comment:
```suggestion
      const icons = await fetch(`${API_URL}/icons`).then(res => res.json()).catch(console.error)
```
Никогда не забывай про catch.
Позже добавим компонент message из Restman.

Reply:
поправила